blivet/partitioning.py

```diff
@@ -681,12 +688,14 @@ def allocate_partitions(storage, disks, partitions, freespace):
             # no disks specified means any disk will do
             req_disks = disks
 
-        # sort the disks, making sure the boot disk is first
         req_disks.sort(key=storage.compare_disks_key)
-        for disk in req_disks:
-            if storage.boot_disk and disk == storage.boot_disk:
-                boot_index = req_disks.index(disk)
-                req_disks.insert(0, req_disks.pop(boot_index))
+
+        if boot_disk:
+            # make sure the boot disk is at the beginning of the disk list
+            for disk in req_disks[:]:
+                if storage.boot_disk and disk == storage.boot_disk:
```

Comment:
You should probably catch `AttributeError` here like we do up in `do_partitioning` since `boot_disk` being not-`None` doesn't  guarantee that storage has a `boot_disk` attribute.